crates/driver/src/tests/setup/mod.rs

```diff
@@ -1,12 +1,988 @@
+use {
+    self::{blockchain::Fulfillment, driver::Driver, solver::Solver},
+    crate::{
+        domain::{competition::order, eth},
+        infra,
+        tests::{
+            cases::{
+                AB_ORDER_AMOUNT,
+                CD_ORDER_AMOUNT,
+                DEFAULT_POOL_AMOUNT_A,
+                DEFAULT_POOL_AMOUNT_B,
+                DEFAULT_POOL_AMOUNT_C,
+                DEFAULT_POOL_AMOUNT_D,
+                DEFAULT_SCORE_MAX,
+                DEFAULT_SCORE_MIN,
+                DEFAULT_SURPLUS_FACTOR,
+                DEFAULT_SURPLUS_FEE,
+                ETH_ORDER_AMOUNT,
+            },
+            setup::blockchain::Blockchain,
+        },
+        util,
+    },
+    ethcontract::BlockId,
+    hyper::StatusCode,
+    secp256k1::SecretKey,
+    std::{
+        collections::{HashMap, HashSet},
+        path::PathBuf,
+        str::FromStr,
+    },
+    web3::types::TransactionId,
+};
 
-pub mod blockchain;
-pub mod driver;
-pub mod solver;
+mod blockchain;
+mod driver;
+mod solver;
 
-pub mod new;
+#[derive(Debug, Clone, Copy)]
+pub struct Asset {
+    token: &'static str,
+    amount: eth::U256,
+}
 
-pub use self::solver::Solver;
+/// Set up a difference between the placed order amounts and the amounts
+/// executed by the solver. This is useful for testing e.g. asset flow
+/// verification. See [`crate::domain::competition::solution::Settlement`].
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct ExecutionDiff {
+    // TODO I think only increase_sell and decrease_buy make sense
+    /// Increase the sell amount executed by the solver by the specified amount.
+    pub increase_sell: eth::U256,
+    /// Decrease the sell amount executed by the solver by the specified amount.
+    pub decrease_sell: eth::U256,
+    /// Increase the buy amount executed by the solver by the specified amount.
+    pub increase_buy: eth::U256,
+    /// Decrease the buy amount executed by the solver by the specified amount.
+    pub decrease_buy: eth::U256,
+}
+
+impl ExecutionDiff {
+    pub fn increase_sell() -> Self {
+        Self {
+            increase_sell: 300.into(),
+            ..Default::default()
+        }
+    }
+
+    pub fn decrease_buy() -> Self {
+        Self {
+            decrease_buy: 300.into(),
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Order {
+    pub name: &'static str,
+
+    /// The amount of the sell token being sold by this order.
+    pub amount: eth::U256,
+    pub sell_token: &'static str,
+    pub buy_token: &'static str,
+
+    pub internalize: bool,
+    pub side: order::Side,
+    pub partial: order::Partial,
+    pub valid_for: util::Timestamp,
+    pub kind: order::Kind,
+
+    // TODO For now I'll always set these to zero. But I think they should be tested as well.
+    // Figure out what (if anything) would constitute meaningful tests for these values.
+    pub user_fee: eth::U256,
+    pub solver_fee: eth::U256,
```

Comment:
The `solver_fee` should only ever affect the `score` of the solution. So you could have a test with 2 identical orders but one has a bigger `solver_fee`. Then the one with the bigger `solver_fee` should also have a higher score.

And for the `user_fee` you could assert that the `user_fee` remains in the settlement contract after settling an order with non-zero `user_fee`.